lib/modules/manager/gomod/artifacts.ts

```diff
@@ -410,3 +413,15 @@ export async function updateArtifacts({
     ];
   }
 }
+
+async function getGoConstraints(
+  goModFileName: string
+): Promise<string | undefined> {
+  const content = (await readLocalFile(goModFileName, 'utf8')) ?? null;
+  if (!content) {
+    return undefined;
+  }
+  const re = regEx(/go\s*(?<gover>\d+.\d+)/);
+  const match = re.exec(content);
+  return match?.groups?.gover;
+}
```

Comment:
This should return a constraint like `^1.17` and not `1.17` (check the deleted extract code)